utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/serde/CirceJsonDeserializer.scala

```diff
@@ -22,8 +24,28 @@ class CirceJsonDeserializer(jsonSchema: Schema) {
   }
 
   def deserialize(string: String): Validated[circe.Error, Any] = {
+    //we basically do parsing twice:
+    //1. for schema validation
+    //2. for typing purposes which is based on Circe
     val jsonObject = new JSONTokener(string).nextValue()
-    jsonSchema.validate(jsonObject)
-    Validated.fromEither(CirceUtil.decodeJson[Json](string).map(JsonToObject.apply(_, swaggerTyped)))
+    //after validate jsonObject has set default field values
+    catchValidationError(jsonSchema.validate(jsonObject))
+    Validated.fromEither(CirceUtil.decodeJson[Json](jsonObject.toString).map(JsonToObject.apply(_, swaggerTyped)))
+  }
+
+  protected def prepareValidationErrorMessage(exception: Throwable): String = {
+    exception match {
+      case ve: ValidationException => ve.getAllMessages.asScala.mkString("\n\n")
+      case je: JSONException => s"Invalid JSON: ${je.getMessage}"
+      case _ => "unknown error message"
```

Comment:
Can we start message from big letter? And also I think it would be nice to see exception message?

Reply:
It's moved code from JsonSchemaRequestResponseSource, i don't want to change de behaviour